Feature/task11 retrieve information for general wrestler table

TiagoVidal91 opened the pull request:
- Creation of endpoint for retuning a list of wrestlers, filtered by lockerroom, brand, number of victories, losses and streaks, and percentage of victories.
- Changes made to the process of fetching data from database (JPA to JDBI)
- Unit Tests
- Made changes to the get JWT method to return null instead of an error, since when the user did not login he would get an error when he wasn't supposed to.
- Changes the Units Tests to reflect this

TiagoVidal91 committed:
TASK 12/13/14/15 - Create Controller/Service/Repository/Entities

- Functionality is capable of getting results by Brand and Lockerrom.

TiagoVidal91 committed:
TASK 12/13/14/15 - Create Controller/Service/Repository/Entities

- Functionality is capable of getting results while ordering them by Wrestler Name, Victories, Losses, Victory Percentage and Streak

TiagoVidal91 committed:
Task 16 - Unit Tests [Controller, Services, Enum]

- Did some additional Code Refactoring
- Corrected some previous tests that stopped working, either for code change, or because the token had expired.

TiagoVidal91 committed:
Task 16 - Unit Tests [Repository]

- Fixed a bug in one of the Service Tests

clara-a-correia commented on src/main/java/com/bearAndPupperCo/sangenWrestlingApp/Services/WrestlerServiceImpl.java:
```diff
@@ -1,106 +1,62 @@
     @Override
     public Wrestler addNewWrestler(Wrestler wrestler) {
-        verifyIfWrestlerExists(wrestler);
+        //verifyIfWrestlerExists(wrestler);
```
add TODO to be able to find - maybe even associate an issue/task number if already existing

TiagoVidal91 replied:
Done!

clara-a-correia commented on src/test/java/com/bearAndPupperCo/sangenWrestlingApp/Repository/WrestlerRepoImplTest.java:
```diff
@@ -0,0 +1,312 @@
+    private String orderDirection = "ASC";
+
+    @Test
+    public void findWrestlerListByRingNameASC() {
```
Simplify methods according to the variable being changed. 
p.ex you can combine findWrestlerListByRingNameASC and DESC, etc 
keep the complexity you find necessary for it to be readable and expandable

TiagoVidal91 replied:
Done

clara-a-correia commented:
Consider changing the methodology of the unit tests for them to act on mock data rather than on production tables. A change in the tables could affect the test results, and your intent should be to test the logic of the code and not the results themselves

TiagoVidal91 committed:
Task 16 - Changes Requested

Made changes to the main query, since it did not return the correct results for the total of victories, total losses and percentage.
Implemented H2 to recriate the database in memory for testing porpuses (Added the data and schema sql for this porpuse)
Made changes to the Unit tests to be less verbose and more efficient
Included more Unit tests that were not considered then (order by total victories, total losses, percentage and streak)
Added a new validation to the findAllWrestlersByParams method on the Service class to verify if the user is sending the correct param to be converted to the table name

TiagoVidal91 committed:
Task 16 - Unit Tests Resources

- Added the necessary resources to run the tests

TiagoVidal91 commented:
- Made changes to the main query, since it did not return the correct results for the total of victories, total losses and percentage.
- Implemented H2 to recriate the database in memory for testing porpuses (Added the data and schema sql for this porpuse)
- Made changes to the Unit tests to be less verbose and more efficient
- Included more Unit tests that were not considered then (order by total victories, total losses, percentage and streak)
- Added a new validation to the findAllWrestlersByParams method on the Service class to verify if the user is sending the correct param to be converted to the table name
